src/components/expression/expression-renderer.js

```diff
@@ -74,99 +35,138 @@ genotet.ExpressionRenderer = function(container, data) {
   this.conditionLabelHeight_ = 0;
 
   /**
-   * The factor of the maximum height of the vertical experiment condition
-   * labels.
+   * Height of the gene profile plot.
+   * This value will be zero when profiles are not shown.
    * @private {number}
    */
-  this.CONDITION_LABEL_HEIGHT_FACTOR_ = 6.501175;
+  this.profileHeight_ = this.DEFAULT_PROFILE_HEIGHT;
 
   /**
-   * The margin between the labels and the heatmap.
+   * Width of the heatmap.
    * @private {number}
    */
-  this.LABEL_MARGIN_ = 10;
+  this.heatmapWidth_ = 0;
 
   /**
-   * The height of the label text.
+   * Height of the heatmap.
    * @private {number}
    */
-  this.TEXT_HEIGHT_ = 14.5;
+  this.heatmapHeight_ = 0;
 
   /**
-   * The size of the gene profile color category.
-   * @private {number}
+   * Clicked object for heatmap cells and gene profiles.
+   * @private {!genotet.ExpressionRenderer.Cell} object
    */
-  this.COLOR_CATEGORY_SIZE_ = 60;
+  this.clickedObject_ = new genotet.ExpressionRenderer.Cell({
+    container: null,
+    geneName: null,
+    conditionName: null,
+    row: -1,
+    column: -1,
+    value: 0
+  });
 
   /**
-   * The default gene profile color category.
-   * @private {array}
+   * Margins of the gene profile.
+   * @private {!Object<number>}
    */
-  this.COLOR_CATEGORY_ = d3.scale.category20()
-    .range()
-    .concat(d3.scale.category20b().range())
-    .concat(d3.scale.category20c().range());
+  this.profileMargins_ = {
+    top: 10,
+    right: 0,
+    bottom: 10,
+    left: 40
+  };
 
   /**
-   * Margins of the gene profile plot.
-   * @private {!Object}
+   * Margins of the heatmap gradient.
+   * @private @const {!Object<number>}
    */
-  this.PROFILE_MARGINS_ = {
-    TOP: 10,
-    RIGHT: 0,
-    BOTTOM: 20,
+  this.HEATMAP_GRADIENT_MARGINS_ = {
+    TOP: 5,
+    RIGHT: 40,
+    BOTTOM: 5,
     LEFT: 40
   };
+};
 
-  /**
-   * Height of the gene profile plot.
-   * This value will be zero when profiles are not shown.
-   * @private {number}
-   */
-  this.profileHeight_ = this.DEFAULT_PROFILE_HEIGHT;
-
-  /**
-   * Width of the heatmap.
-   * @private {number}
-   */
-    this.heatmapWidth_ = 0;
-  /**
-   * Height of the heatmap.
-   * @private {number}
-   */
-  this.heatmapHeight_ = 0;
+genotet.utils.inherit(genotet.ExpressionRenderer, genotet.ViewRenderer);
 
-  /**
-   * Margin of the gene profile.
-   * @private {number}
-   */
-  this.geneProfileMargin_ = 0;
+/**
+ * Cell object storing the rendering properties of expression cell.
+ * @struct
+ * @constructor
+ * @param {!Object} params
+ */
+genotet.ExpressionRenderer.Cell = function(params) {
+  this.container = params.container;
+  this.geneName = params.geneName;
+  this.conditionName = params.conditionName;
+  this.row = params.row;
```

Comment:
If you have default values you can say 
```js
this.row = params.row != null ? params.row : -1;
```
so that you don't need to pass in default values for every constructed objects.